src/material-experimental/mdc-checkbox/checkbox.ts

```diff
@@ -77,6 +77,8 @@ const _MatCheckboxBase = mixinColor(
   host: {
     'class': 'mat-mdc-checkbox',
     '[attr.tabindex]': 'null',
+    '[class.mat-mdc-checkbox-disabled]': 'disabled',
+    '[class.mat-mdc-checkbox-checked]': 'checked',
```

Comment:
Since these classes aren't used anywhere, should we add a note about why they're being assigned?